lib/src/git.rs

```diff
@@ -60,6 +60,38 @@ fn local_branch_name_to_ref_name(branch: &str) -> String {
     format!("refs/heads/{branch}")
 }
 
+fn resolve_git_ref_to_commit_id(
```

Comment:
Minor, optional or feel free to adapt however convenient: I would add a description here:

```
/// This function will return the same commit as `git_ref.peel_to_commit()`,but is faster
// in the common case where `git_ref` points to the same commit as `known_target`.
```

I think that also makes the `/// Try fast path..` comment (which I found a bit confusing) unnecessary.